### DISCUSSION TITLE
关于国内用户无法从 GitHub 安装博客主题的一点愚见

### DISCUSSION THREAD

**Discussion by chuxinyuan** [rbind/yihui#1520] [General]
当年我就卡在不能访问 GitHub 这里好久，终于在修改 DNS、devsidecar 和 Gitee 多管齐下才手动解决了问题。为了避免更多的的新手卡在不能访问 GitHub 这里，我想了两个解决方案：

一种解决方案是把博客主题同步一个副本到 GitLab 上，让用户选择勾选 GitHub 或者 GitLab；

方案二是 `install_theme` 函数里面做一个判断，如果一定时间内没有连接到 GitHub，则使用代理从 GitHub 仓库下载文件。我用了一段时间 [GitHub Proxy](https://ghproxy.com/)，比较稳定，速度也很快。

**Comment by yihui** [comment]
> 方案二是 `install_theme` 函数里面做一个判断

应该不用这么费劲，自定义 `hostname` 参数即可：

```r
blogdown::install_theme(hostname = 'ghproxy.com/https://github.com')
```

**Reply by chuxinyuan** [reply]
方案一应该更丝滑点。

**Comment by chuxinyuan** [comment]
各位看官，<https://ghproxy.com/> 貌似挂了，不过 <https://gh-proxy.com/> 暂时能用。
